skyportal/handlers/api/invitations.py

```diff
@@ -0,0 +1,143 @@
+import uuid
+import python_http_client.exceptions
+from baselayer.app.access import permissions
+from baselayer.app.env import load_env
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    Group,
+    GroupStream,
+    Invitation,
+    Stream,
+)
+
+_, cfg = load_env()
+
+
+class InvitationHandler(BaseHandler):
+    @permissions(["Manage users"])
+    def post(self):
+        """
+        ---
+        description: Invite a new user
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  userEmail:
+                    type: string
+                  streamIDs:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of IDs of streams invited user will be given access to. If
+                      not provided, user will be granted access to all streams associated
+                      with the groups specified by `groupIDs`.
+                  groupIDs:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of IDs of groups invited user will be added to. If `streamIDs`
+                      is not provided, invited user will be given accesss to all streams
+                      associated with the groups specified by this field.
+                  groupAdmin:
+                    type: array
+                    items:
+                      type: boolean
+                    description: |
+                      List of booleans indicating whether user should be granted admin
+                      status for respective specified group(s).
+                required:
+                  - userEmail
+                  - groupIDs
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+        """
+        if not cfg["invitations.enabled"]:
+            return self.error("Invitations are not enabled in current deployment.")
+        data = self.get_json()
+
+        if data.get("userEmail") is None or data.get("userEmail") in [
+            "",
+            "None",
+            "null",
+        ]:
+            return self.error("Missing required parameter `userEmail`")
+        user_email = data["userEmail"].strip()
+
+        if not data.get("groupIDs") is not None:
```

Comment:
is None?

Reply:
Yes of course, thanks, this will be updated (changed multiple times and came out in bad shape 😂)

Reply:
Done